File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int curPos = nums.length-1;
        for(int i=nums.length-2;i>=0;i--){
            if(i+nums[i]>=curPos)
                {
                    curPos=i;
                }
        }
        if(curPos==0){
            return true;
        }
        return false;
    }
}

File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        if len(nums) == 1:
            return True
        farthest= 0;
        for i in range(len(nums)-1):
            if i > farthest:
                return False;
            
            farthest = max(farthest,i+nums[i]);
            if farthest >= len(nums)-1:
                return True;
        return False;
        